### PR TITLE
Issue #745, adds more options to azurerm_function_app

### PR DESCRIPTION
**Pull request by cloudify**
@tombuildsstuff how do you suggest I test this patch? should I create new tests or extend the current ones?

### CODE REVIEW

File: azurerm/resource_arm_function_app.go [hunk 1]
```diff
@@ -134,6 +144,16 @@ func resourceArmFunctionApp() *schema.Resource {
 							Optional: true,
 							Default:  false,
 						},
+						"use_32_bit_worker_process": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app.go]: Based on [this comment](https://github.com/terraform-providers/terraform-provider-azurerm/issues/745#issuecomment-360933129) I believe this needs to be defaulted to `true` for Function Apps?

File: website/docs/r/function_app.html.markdown [hunk 2]
```diff
@@ -87,7 +89,12 @@ The following arguments are supported:
 
 `site_config` supports the following:
 
-* `always_on` - (Optional) Should the app be loaded at all times? Defaults to `false`.
+* `always_on` - (Optional) Should the Function App be loaded at all times? Defaults to `false`.
+* `use_32_bit_worker_process` - (Optional) Should the Function App run in 32 bit mode, rather than 64 bit mode?
```
Comment (tombuildsstuff) [inline website/docs/r/function_app.html.markdown]: We should add "Defaults to `value`" (which should be true, based on above)

File: azurerm/resource_arm_function_app.go [hunk 3]
```diff
@@ -410,6 +441,14 @@ func flattenFunctionAppSiteConfig(input *web.SiteConfig) []interface{} {
 		result["always_on"] = *input.AlwaysOn
 	}
 
+	if input.Use32BitWorkerProcess != nil {
+		result["use_32_bit_worker_process"] = *input.Use32BitWorkerProcess
+	}
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app.go]: could we add an acceptance test with the 64 bit worker process enabled? here's [an example of how we're doing that](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_function_app_test.go#L14)

Comment (tombuildsstuff) [reply azurerm/resource_arm_function_app.go]: so that we can get this merged, I've pushed an acceptance test for this - I hope you don't mind!

Comment (cloudify) [reply azurerm/resource_arm_function_app.go]: not at all! thanks for pushing this, I got caught in higher priority tasks and couldn't focus on this now...

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2018-03-06 at 09 25 55

**Pull request merged**